packages/logs/src/domain/logsCollection/logger/loggerCollection.ts

```diff
@@ -0,0 +1,46 @@
+import { includes, display, combine, ErrorSource, isExperimentalFeatureEnabled } from '@datadog/browser-core'
+import type { CommonContext } from '../../../rawLogsEvent.types'
+import type { LifeCycle } from '../../lifeCycle'
+import { LifeCycleEventType } from '../../lifeCycle'
+import type { Logger, LogsMessage } from '../../logger'
+import { StatusType, HandlerType } from '../../logger'
+
+export const STATUS_PRIORITIES: { [key in StatusType]: number } = {
+  [StatusType.debug]: 0,
+  [StatusType.info]: 1,
+  [StatusType.warn]: 2,
+  [StatusType.error]: 3,
+}
+
+export function startLoggerCollection(lifeCycle: LifeCycle) {
+  function handleLog(logsMessage: LogsMessage, logger: Logger, savedCommonContext?: CommonContext) {
+    const messageContext = logsMessage.context
+
+    if (isAuthorized(logsMessage.status, HandlerType.console, logger)) {
+      display.log(`${logsMessage.status}: ${logsMessage.message}`, combine(logger.getContext(), messageContext))
+    }
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏at first glance I expected that the `display.log` call should be done in another module using `lifeCycle.subscribe(LOG_COLLECTED, ...)`. But:

* it seems that LOG_COLLECTED (and the whole LifeCycle, and the assembly module) only expects to handle logs that should be sent to datadog. It's a bit unexpected but seems reasonable.

* I think that we should keep the `display.log` call close to the actual `Logger.*` usage, i.e. not moving it in a LifeCycle `subscribe` callback, but maybe moving it back to the `Logger` class. With #1501 we'll have a stack trace displayed below "error" and "warn" logs, which is great! But having `display.log` deep into the stack trace adds a bit of noise and impacts the developper experience. For example, with your PR, the stack trace would look like this:
![Screenshot 2022-04-22 at 17 56 38](https://user-images.githubusercontent.com/61787/164752310-679aaeac-d493-434a-8005-f945c5e67f1d.png)
Which is not dramatic, but we might want to minimize this stack trace :)

Reply:
Yes right. Since this PR is quite big and it takes times to validate, I'm in favor of working on this improvement while doing https://github.com/DataDog/browser-sdk/pull/1501